server/src/main/java/org/opensearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -154,6 +155,14 @@ public ActionRequestValidationException validate() {
         return validationException;
     }
 
+    @Override
+    public List<String> requiredPermissions() {
+        return List.of(
+            "opensearch.engine.index.create",
```

Comment:
This is a good place to start and should be expandable to index patterns. Are you thinking that all permissions in `core` be preceded with `opensearch.engine` and there will be a separate convention for actions added by plug-ins or modules?

Reply:
FYI @shanilpa @jimishs. 

For this specific request when it comes to indexing, its clearly part of the core/engine, or maybe the prefix should just be `opensearch.indexing.index.create`.  Maybe of the nature `[Source].[Category].[Object].[Action]`?

I don't plan on iterating on this name for this pull request and we will need to follow up in the broader vNext permissions conversation.

Reply:
Should these be loaded from a file or some other store? (Could be part of next iteration)

Reply:
I looked into the relationship between permissions and actions and there is no good mechanism 
to engage with actions in a way that produces a list of permissions.

Exploring the ActionModule there is a common system to register actions however this registration doesn't pass all of the information about action execution, it passes a limited version.

What I would recommend would be creating a new description for action related data that encompasses the action type, action request, action response, and the action handler.   Building this into or around the ActionModule seems appropriate but this will need to be rolled out over time since it would be backwards incompatible to plugins.  This is a good follow up item.